File: n520-detectCapitalUse.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function(word) {
    var capCount = 0
    for (var i = 0; i < word.length; i++) {
        if ('Z'.charCodeAt(0) - word[i].charCodeAt(0) >= 0) {
            capCount++
        }
    }
    
    if (capCount === word.length || 
        capCount === 0 ||
        (capCount === 1 && 'Z'.charCodeAt(0) - word[0].charCodeAt(0) >= 0)
    ) {
        return true
    }
    
    return false
};

File: n520-detectCapitalUse.go
package main

func detectCapitalUse(word string) bool {
	data := []rune(word)

	capCount := 0
	for _, c := range data {
		if 'Z'-c >= 0 {
			capCount++
		}
	}

	if capCount == len(data) ||
		capCount == 0 ||
		(capCount == 1 && 'Z'-data[0] >= 0) {
		return true
	}

	return false
}
